torch/csrc/autograd/python_variable.cpp

```diff
@@ -967,7 +967,7 @@ static PyObject* THPVariable_dtensor_new(
   Tensor tensor = make_tensor_for_subclass_helper(
       /*sym_sizes=*/tuple_to_symintlist(sizes.ptr()),
       /*sym_strides=*/tuple_to_symintlist(stride.ptr()),
-      /*sym_storage_offset=*/std::nullopt,
+      /*sym_storage_offset=*/local_tensor.sym_storage_offset(),
```

Comment:
~~this will set the inner local tensor and outer DTensor to always have the same storage offset. Is that what we want? I'm probably missing context but I would have guessed that we would need to stash "global storage offset" separately from the local storage offset.~~

nvm I see your description:
```
The storage offset always matches the local offset because you never have rank dependent offset (your shard may be different, but your view into it will always be the same across all ranks!)
```

Reply:
Yeah originally I thought I needed to compute this the same way size/stride is done, but I thought about it hard and concluded they should always match.